src/main/java/org/pvv/shufflegene/Traverse.java

```diff
@@ -0,0 +1,163 @@
+package org.pvv.shufflegene;
+
+import org.javatuples.Pair;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Representation of a nucleotide sequence as a graph with vertices spanning different bases.
+ */
+public class Traverse {
+    public char start;
+    public char end;
+    // start, end and path of original sequence
+    public int length;
+    private HashMap<Character, ArrayList<Character>> edgeMap;
```

Comment:
I've introduced an edge class, which feels a good way to represent the graph for now.

I'll hold off on swapping chars for nucleotides and hashmaps for objects just now, as I think I might generalize this to other alphabets shortly. Agreed that it's redundant at the moment, since we only allow `ACGT`.